userspace/libsinsp/utils.cpp

```diff
@@ -1920,61 +1920,104 @@ std::set<T> unordered_set_to_ordered(std::unordered_set<T> unordered_set)
 	}
 	return s;
 }
-template std::set<uint32_t> unordered_set_to_ordered(std::unordered_set<uint32_t> unordered_set);
-template std::set<std::string> unordered_set_to_ordered(std::unordered_set<std::string> unordered_set);
+template std::set<uint32_t> unordered_set_to_ordered(const std::unordered_set<uint32_t>& unordered_set);
+template std::set<std::string> unordered_set_to_ordered(const std::unordered_set<std::string>& unordered_set);
 
-// convert_vector_to_unordered_set
+// unordered_set_difference, equivalent to SQL left_anti join operation
 template<typename T>
-std::unordered_set<T> convert_vector_to_unordered_set(std::vector<T> v)
+std::unordered_set<T> unordered_set_difference(const std::unordered_set<T>& a, const std::unordered_set<T>& b)
 {
 	std::unordered_set<T> s;
-	for (const auto &x : v)
+	for(const auto& val : a)
 	{
-		s.insert(x);
+		if (b.find(val) == b.end())
+		{
+			s.insert(val);
+		}
 	}
 	return s;
 }
-template std::unordered_set<std::string> convert_vector_to_unordered_set(std::vector<std::string> v);
-template std::unordered_set<uint32_t> convert_vector_to_unordered_set(std::vector<uint32_t> v);
+template std::unordered_set<std::string> unordered_set_difference(const std::unordered_set<std::string>& a, const std::unordered_set<std::string>& b);
+template std::unordered_set<uint32_t> unordered_set_difference(const std::unordered_set<uint32_t>& a, const std::unordered_set<uint32_t>& b);
 
-// unordered_set_difference, equivalent to SQL left_anti join operation
+// set_difference, equivalent to SQL left_anti join operation
 template<typename T>
-std::unordered_set<T> unordered_set_difference(std::unordered_set<T> a, std::unordered_set<T> b)
+std::set<T> set_difference(const std::set<T>& a, const std::set<T>& b)
 {
-	std::set<T> c = unordered_set_to_ordered(a);
-	std::set<T> d = unordered_set_to_ordered(b);
-	std::vector<T> v;
-	std::set_difference(c.begin(), c.end(), d.begin(), d.end(), std::inserter(v, v.begin()));
-	return convert_vector_to_unordered_set(v);
+	std::set<T> s;
```

Comment:
For ordered `set`, I think we could rely on https://cplusplus.com/reference/algorithm/set_difference/ right away, which should also be more efficient.